### PR TITLE
Add EntityContextMenu export to index.ts

### PR DESCRIPTION
**Pull request by LindaL2860**
## Hey, I just made a Pull Request!

Added the EntityContextMenu to the export for having it as an import in our code instead of copying the code.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/catalog/api-report.md [hunk 1]
```diff
@@ -393,10 +401,10 @@ export interface EntityLayoutProps {
   //
   // (undocumented)
   UNSTABLE_contextMenuOptions?: EntityContextMenuOptions;
-  // Warning: (ae-forgotten-export) The symbol "ExtraContextMenuItem" needs to be exported by the entry point index.d.ts
+  // Warning: (ae-forgotten-export) The symbol "ExtraContextMenuItem_2" needs to be exported by the entry point index.d.ts
```
Comment (nuritizra) [inline plugins/catalog/api-report.md:404]: can you please explain the change here? why did you add _2?

Comment (freben) [reply plugins/catalog/api-report.md:404]: This is just an artifact of how these machine generated api reports get produced by the scripts. You can double check at the top of the file what import it moved from/to.

**Review by nuritizra — APPROVED** [review]

File: plugins/catalog/api-report.md [hunk 2]
```diff
@@ -330,6 +330,14 @@ export interface DependsOnResourcesCardProps {
 // @public (undocumented)
 export const EntityAboutCard: (props: AboutCardProps) => JSX.Element;
 
+// Warning: (ae-forgotten-export) The symbol "EntityContextMenuProps" needs to be exported by the entry point index.d.ts
+// Warning: (ae-missing-release-tag) "EntityContextMenu" is part of the package's API, but it is missing a release tag (@alpha, @beta, @public, or @internal)
```
Comment (freben) [inline plugins/catalog/api-report.md:334]: Note these two warnings. Especially the one about types. One of the reasons that this one wasn't exported so far, is that it has a pretty complex set of type dependencies. When you import this component you won't be getting proper types for its props. We also would probably like to NOT export all of those types because it expands the footprint of this package with noise.

Let's merge for now, but if this proves problematic, it would be preferable to _inline_ the types used in the props by hand, eg rewriting the menu as 

```ts
export function EntityContextMenu(props: {
  UNSTABLE_extraContextMenuItems?: Array<{
    title: string;
    Icon: IconComponent;
    onClick: () => void;
  }>;
  UNSTABLE_contextMenuOptions?: {
    disableUnregister: boolean | 'visible' | 'hidden' | 'disable';
  };
  onUnregisterEntity: () => void;
  onInspectEntity: () => void;
})
```

**Review by freben — APPROVED** [review]
Not a great fit for exports - but OK, I realize that this is basically the only entrypoint for un-registration at this point

**Comment by freben** [comment]
Oh. Before we merge. I see that the DCO check isn't green. That's important - check the check details or the bot comment above about how to re-sign your commits. Every one needs to be signed. Sorry about the trouble, it's for license reasons.

**Comment by freben** [comment]
I actually can't, github doesn't allow it at this point.

**Comment by LindaL2860** [comment]
So i do need to make a new PR?

**Comment by freben** [comment]
Yes, we can't really do anything on our side I think.

**Comment by LindaL2860** [comment]
Thank you @freben, and thanks for your patience.
The new PR can be found at https://github.com/backstage/backstage/pull/25086
